stable/kong/templates/service-kong-proxy.yaml

```diff
@@ -25,13 +25,45 @@ spec:
   {{- end }}
   {{- end }}
   ports:
+  {{- if and .Values.proxy.http.enabled .Values.proxy.tls.enabled }}
   - name: kong-proxy
-    port: {{ .Values.proxy.servicePort }}
-    targetPort: {{ .Values.proxy.containerPort }}
-  {{- if (and (eq .Values.proxy.type "NodePort") (not (empty .Values.proxy.nodePort))) }}
-    nodePort: {{ .Values.proxy.nodePort }}
+    port: {{ .Values.proxy.http.servicePort }}
+    targetPort: {{ .Values.proxy.http.containerPort }}
+  {{- if (and (eq .Values.proxy.type "NodePort") (not (empty .Values.proxy.http.nodePort))) }}
+    nodePort: {{ .Values.proxy.http.nodePort }}
   {{- end }}
-    protocol: TCP
+  - name: kong-proxy-tls
+    port: {{ .Values.proxy.tls.servicePort }}
+    targetPort: {{ .Values.proxy.tls.containerPort }}
+  {{- if (and (eq .Values.proxy.type "NodePort") (not (empty .Values.proxy.tls.nodePort))) }}
+    nodePort: {{ .Values.proxy.tls.nodePort }}
+  {{- end }}
+
+  {{ else }}
+
+  {{- if .Values.proxy.http.enabled }}
+  - name: kong-proxy
+    port: {{ .Values.proxy.http.servicePort }}
+    targetPort: {{ .Values.proxy.http.containerPort }}
+  {{- if (and (eq .Values.proxy.type "NodePort") (not (empty .Values.proxy.http.nodePort))) }}
+    nodePort: {{ .Values.proxy.http.nodePort }}
+  {{- end }}
+  {{- end }}
+  {{- if or .Values.proxy.tls.enabled .Values.proxy.tls.enableTlsOffload }}
```

Comment:
@hbagdi I neded one special case here to point this to the http `containerPort` instead of the tls one.

Reply:
You can do that in the values.yaml by changing the `proxy.tls.containerPort` to be same as `proxy.http.containerPort`, right?

Reply:
This entire section will only output on this if statement:  `{{- if .Values.proxy.http.enabled }}`

If that is set to true then in the proxy deployment file it will set the `KONG_PROXY_LISTEN` to be enabled also.

Now thinking about it.  Why not just enable it and then do what you suggest by just pointing the port to the non tls port.  Im ok with that route also.

Reply:
Yes, let's just do the part that I suggested. It will be maintainable for us and flexible and easy for the end user.

Reply:
If i use the same port i get a string like this for the listen address.  Kong fails to start with the overlapping ports

```
        - name: KONG_PROXY_LISTEN
          value: 0.0.0.0:8000,0.0.0.0:8000 ssl
```

Any suggestions?

Reply:
@hbagdi ?

Reply:
Can we add support for only TLS and HTTP at the same time in this PR and then visit the TLS offload thing in another separate PR? It will make reviewing this code and getting it merged in faster.

Also, any reason why it's a problem that ELB can't talk to Kong's TLS port even when it is doing the TLS termination? IIRC ELB doesn't verify the cert of your backend server.

Reply:
Sure, let me separate it out into two different PRs.

It would set up a topology describe here (https://github.com/helm/charts/pull/10003/files#diff-bd36c84be1fb63f9d09295a16068a4a4R195).  Hmm...I believe when I tried, it was because kong had a self signed cert and the ELB wouldnt connect.  I am going to reduce the scope of this PR and then will test what you asked about why dont the ELB terminate to kong on a TLS port.

Reply:
@hbagdi here is the new PR:  https://github.com/helm/charts/pull/10182

Closing this one.